src/ui/page/share/dataset/steps/metadataFilling/metadataFilling.tsx

```diff
@@ -318,26 +366,49 @@ export const MetadataFilling: FC = () => {
         }
       }
     ]
-  }, [addTag, formItemFieldValue, handleFieldValueChange, removeTag, t, tagsFieldValue])
+  }, [t, handleFieldValueChange, singleValueField, multiValuesField, handleNumericValueChange])
 
-  const mapForm = (form: DatasetForm): initFormPayload => {
-    return form.map(formItem => ({
-      id: formItem.id,
-      type: formItem.type,
-      label: formItem.label,
-      value: formItem.value,
-      required: formItem.required
-    }))
-  }
+  const mapForm = (form: DatasetForm): InitFormPayload =>
+    form.map(formItem => {
+      const typedFormItem = {
+        id: formItem.id,
+        title: formItem.title,
+        required: formItem.required,
+        type: formItem.type,
+        value: formItem.value
+      }
+
+      switch (formItem.type) {
+        case 'i18n-text':
+          return typedFormItem as I18NTextField
+
+        case 'text':
+          return typedFormItem as TextField
+
+        case 'numeric':
+          return typedFormItem as NumericField
+
+        case 'select':
+          return typedFormItem as SelectPicker
+
+        case 'tag':
+          return typedFormItem as TagField
+      }
+    })
```

Comment:
Since the form types are already a discriminated union using the type property, you can take advantage of this for better type checking:
```suggestion
const mapForm = (form: DatasetForm): InitFormPayload => {
    return form.map(formItem => {
      switch (formItem.type) {
        case 'i18n-text':
          return {
            ...formItem,
            type: 'i18n-text'
          }
        case 'text':
          return {
            ...formItem,
            type: 'text'
          }
        case 'numeric':
          return {
            ...formItem,
            type: 'numeric'
          }
        case 'select':
          return {
            ...formItem,
            type: 'select'
          }
        case 'tag':
          return {
            ...formItem,
            type: 'tag'
          }
      }
    })
  }
```

Reply:
yes, working too! and working even if I just return the `formItem`, auto typed thanks to the `case`, but looking strange ;)